### ISSUE TITLE
"message":"not supported" issue.

### ISSUE THREAD

**Issue by PritamUpadhyay** [mcollina/mosca#374]
````javascript
{
"name":"mosca","hostname":"DELLNCL-W2012","pid":4688,"client":"pri9e8423bab3cc3a41/00158d000052b96a","level":40,"err":{"message":"not supported","name":"Error","stack":"Error: not supported\n
at Parser._parsePayload (C:\\Program Files\\nodejs\\node_modules\\mosca\\node_modules\\mqtt-connection\\node_modules\\mqtt-packet\\parser.js:142:28)\n
at Parser.parse (C:\\Program Files\\nodejs\\node_modules\\mosca\\node_modules\\mqtt-connection\\node_modules\\mqtt-packet\\parser.js:42:48)\n
at DestroyableTransform.process [as _transform] (C:\\Program Files\\nodejs\\node_modules\\mosca\\node_modules\\mqtt-connection\\lib\\parseStream.js:13:12)\n
at DestroyableTransform.Transform._read (C:\\Program Files\\nodejs\\node_modules\\mosca\\node_modules\\mqtt-connection\\node_modules\\through2\\node_modules\\readable-stream\\lib\\_stream_transform.js:184:10)\n
at DestroyableTransform.Transform._write (C:\\Program Files\\nodejs\\node_modules\\mosca\\node_modules\\mqtt-connection\\node_modules\\through2\\node_modules\\readable-stream\\lib\\_stream_transform.js:172:12)\n
at doWrite (C:\\Program Files\\nodejs\\node_modules\\mosca\\node_modules\\mqtt-connection\\node_modules\\through2\\node_modules\\readable-stream\\lib\\_stream_writable.js:237:10)\n
at writeOrBuffer (C:\\Program Files\\nodejs\\node_modules\\mosca\\node_modules\\mqtt-connection\\node_modules\\through2\\node_modules\\readable-stream\\lib\\_stream_writable.js:227:5)\n
at DestroyableTransform.Writable.write (C:\\Program Files\\nodejs\\node_modules\\mosca\\node_modules\\mqtt-connection\\node_modules\\through2\\node_modules\\readable-stream\\lib\\_stream_writable.js:194:11)\n
at write (_stream_readable.js:583:24)\n    at flow (_stream_readable.js:592:7)\n
at Socket.pipeOnReadable (_stream_readable.js:624:5)\n
at Socket.EventEmitter.emit(events.js:92:17)"},"msg":"not supported","time":"2015-11-21T11:48:20.014Z","v":0
}
````

**Comment by mcollina** [comment]
That happens when the packet is malformed, or we might have a bug in the parsing logic. Can you upload a test to replicate?

**Comment by mmick66** [comment]
I got the same error just now. I am going to try and replicate.

**Comment by mcollina** [comment]
It should be fixed in v1.0.0, if that ever was an issue

**Comment by PritamUpadhyay** [comment]
Hi. This is happening again in new version. Any work around for this issue?
My current version is 1.4.0

**Comment by Travis346** [comment]
I am consistently getting this error as well (v2.1.0).  A malformed packet makes sense as I sometimes see erroneous characters arrive when receiving packets in quick succession.  Is there a clean way to catch this error (disregard the malformed packet) such that it doesn't crash the server?

**Comment by mcollina** [comment]
@Travis346 can you reproduce it from a Node.js script, so that I can reproduce it here? Otherwise it will be extremely hard to solve.

A PR is always welcomed.

**Comment by Travis346** [comment]
The issue occurs using the embedded Mosca quick start example with no backend.  The client I am using is a ESP8266-12E Arduino core with the following MQTT library: [https://github.com/knolleary/pubsubclient](url)
I modified the example code 'mqtt_esp8266.ino'  to send a message on every loop like the following (c++ code):

`void loop() {
  client.loop();
  Serial.println(msg);
  client.publish("outTopic", msg);
}`

This works fine for at least 100 or so messages but then stops the server as described above.  I have repeated this issue with different microcontrollers and various MQTT package.  I believe the malformed messages are being generated by these cheap WIFI modules when sending in quick succession.  The ESP8266 doesn't seem to notice that it sent a bad package and keeps running but it unfortunately crashes Mosca.   

I'll try and implement a catch for this type of error and submit a PR if successful.
